### DISCUSSION TITLE
API terms of service: Section 5.e.1. Prohibition on building databases

### DISCUSSION THREAD

**Discussion by ryan-lp** [Podcastindex-org/podcast-namespace#558] [Q&A]
[Podcast Index APIs Terms of Service - March 2, 2021](https://api.podcastindex.org/tos_v1.1.pdf)

Reading Section 5.e.,

[you will not]:

1. Scrape, build databases, or otherwise create permanent copies of such content, or keep cached copies longer than permitted by the cache header;

It seems from this that the way I would want to use the API is actually not permitted, and that's why I have instead built a sync tool to import the weekly data dumps, compute a diff of what's changed, and then replicate all of the APIs that I need. I understand that the weekly data dump is there to prevent scraping the API, but it's not clear to me why someone who already has a copy of the data dump shouldn't also be able to make occasional API calls and integrate those responses into a local copy of the database.

Prohibition 2 is also a concern to the extent that it relates to 1.:

2. Copy, translate, modify, create a derivative work of, sell, lease, lend, convey, distribute, publicly display, or sublicense to any third-party;

I am also curious to ask, what do the cache headers referenced in prohibition 1 actually say?

As for what I'm currently doing (downloading the whole database every week, computing a diff, then integrating that), I feel this could be streamlined. Currently, it's a large file to download, it takes time for the diff/import process to complete, and I don't think the dump is at a predictable URL so it's a process I have to do manually. It would perhaps be more convenient and efficient to publish diffs. Or, and maybe you have reasons against this, since the Podcast Index actually knows when all podcasts get updated, it could just podping the changes on behalf of those that don't already use podping.

**Comment by daveajones** [comment]
We don’t care what you do with the data. If you want to build a database knock yourself out. Even commercial is fine. The biggest issue we have is API scraping which is what we combat constantly. I’m fine with people building databases but I don’t want them using the API to do it (meaning issuing 100 million requests just to get all the episodes).  I’d rather provide a download.  It’s just that the full (with episodes) download is 150 gigs so it’s why I only provide the feeds tabs currently. 

I’m not sure what you mean by the weekly dump “not being at a predictable url”. The download url for that hasn’t changed in a long time. If there is a bug let me know.

What @mitchdowney does for Podverse is he tracks the /recent/data endpoint and just follows along with all the feed updates. There are a bunch of ways to stay up to date actually.  I’m glad to share them. 

Sorry for the late reply. This should probably be in the “database” repo instead of in the namespace repo.

**Reply by ryan-lp** [reply]
Or maybe the "api" repo? The database repo seems to be about the design of the software, not about the API terms and conditions. Sorry, I agree I perhaps did not choose the best one. (I found a page on how to transfer it [here](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#transferring-a-discussion) if you would like to relocate this to somewhere more appropriate)

I understand that you probably wouldn't have a problem with building databases, but since the terms of service explicitly prohibit building databases, I thought I would raise the issue since it is the reason why I am using the dumps instead of the API.

> The download url for that hasn’t changed in a long time

Don't worry, I think my comment was just out of date, that issue is since resolved.

**Reply by ryan-lp** [reply]
> Or maybe the "api" repo?

I just found there is a repo called "[legal](https://github.com/Podcastindex-org/legal)" which hosts the terms of service document, so maybe I should have opened an issue there.

**Reply by ryan-lp** [reply]
> Sorry, I agree I perhaps did not choose the best one.

I have moved the issue to https://github.com/Podcastindex-org/legal/issues/1

**Comment by Charlotte-br560** [comment]
Understand the frustration with API terms. Your workaround with weekly data dumps sounds smart! Have you considered exploring publishing differentials or utilizing podping for updates?
